java/app/src/main/java/com/google/android/gms/samples/wallet/activity/PaymentTransparentActivity.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2020 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gms.samples.wallet.activity;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.widget.Toast;
+
+import com.google.android.gms.samples.wallet.util.Notifications;
+import com.google.android.gms.samples.wallet.util.PaymentsUtil;
+import com.google.android.gms.samples.wallet.R;
+import com.google.android.gms.wallet.AutoResolveHelper;
+import com.google.android.gms.wallet.PaymentData;
+import com.google.android.gms.wallet.PaymentDataRequest;
+import com.google.android.gms.wallet.PaymentsClient;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Optional;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+/**
+ * Transparent activity to be triggered from notification
+ */
+public class PaymentTransparentActivity extends AppCompatActivity {
+
+  // Arbitrarily-picked constant integer you define to track a request for payment data activity.
+  private static final int LOAD_PAYMENT_DATA_REQUEST_CODE = 991;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_transparent);
+
+    // Dismiss the notification UI if the activity was opened from a notification
+    if (Notifications.ACTION_PAY_GOOGLE_PAY.equals(getIntent().getAction())) {
+      sendBroadcast(new Intent(Intent.ACTION_CLOSE_SYSTEM_DIALOGS));
```

Comment:
Do you need a broadcast to dismiss the notification? Isn't enough using the notification id (if you know it)

Reply:
This does not remove the notification, but dismissed the entire notification panel/view on Android